examples/ibm-power/main.tf

```diff
@@ -34,16 +45,18 @@ resource "ibm_pi_volume" "volume" {
 data "ibm_pi_volume" "data_source_volume" {
   depends_on = [ibm_pi_volume.volume]
 
-  pi_cloud_instance_id = var.cloud_instance_id
+  pi_cloud_instance_id = ibm_pi_workspace.powervs_service_instance.id
   pi_volume_name       = var.volume_name
 }
+
+# Create an instance
 resource "ibm_pi_instance" "instance" {
   depends_on = [data.ibm_pi_image.data_source_image,
     data.ibm_pi_key.data_source_key,
     data.ibm_pi_volume.data_source_volume,
-  data.ibm_pi_public_network.data_source_network]
+  data.ibm_pi_network.data_source_private_network]
 
```

Comment:
depends_on are not required if the element is used, terraform calculates the dependency and guarantees the order.  For example depends_on data.ibm_pi_key.data_source_key is not need needed since data.ibm_pi_key.data_source_key.id is used below.

Reply:
If starting from scratch on a new workspace, I definitely got errors removing the dependency statements. On the image/network creation if I tried to read from the data source right after.

Reply:
Changed code to remove depends on and use the resources to let terraform figure out the dependencies.